File: app/src/main/java/com/example/todaysshow/SearchQueryAdapter.java
package com.example.todaysshow;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Filter;
import android.widget.Filterable;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class SearchQueryAdapter extends RecyclerView.Adapter<SearchQueryAdapter.SearchQueryItemView> implements Filterable {

    public interface OnItemClickListener{
        void onItemClick(View v, int pos);
    }
    private OnItemClickListener mListener = null;

    public void setOnItemClickListener(OnItemClickListener listener){
        this.mListener = listener;
    }

    ArrayList<String> searchList;
    Context context;
    ArrayList<String> filteredList;
    ArrayList<String> unFilteredList;

    SearchQueryAdapter(ArrayList<String> searchList, Context context){
        this.searchList = searchList;
        this.unFilteredList = searchList;
        this.filteredList = searchList;
        this.context = context;
    }

    @NonNull
    @Override
    public SearchQueryItemView onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.search_list_item, parent, false);
        return new SearchQueryItemView(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull SearchQueryItemView holder, int position) {
        final String searchStr = filteredList.get(position);
        holder.search_qeury_tv.setText(searchStr);
        //holder.journal_iv.setImageResource(journal.getImageResource());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(context, searchStr , Toast.LENGTH_SHORT).show();

            }
        });

    }

    @Override
    public int getItemCount() {
        return filteredList.size();
    }


    @Override
    public Filter getFilter() {
        return new Filter() {
            @Override
            protected FilterResults performFiltering(CharSequence constraint) {
                String charString = constraint.toString();
                if(charString.isEmpty()){
                    filteredList = unFilteredList;
                }else{
                    ArrayList<String> filteringList = new ArrayList<>();
                    for(String name : unFilteredList){
                        if(name.toLowerCase().contains(charString.toLowerCase())){
                            filteringList.add(name);
                        }
                    }
                    filteredList = filteringList;
                }
                for(String name : filteredList){
                    Log.i("Filtered", name);
                }
                FilterResults filterResults = new FilterResults();
                filterResults.values = filteredList;
                return filterResults;
            }

            @Override
            protected void publishResults(CharSequence constraint, FilterResults results) {
                filteredList = (ArrayList<String>)results.values;
                notifyDataSetChanged();
            }
        };
    }

    public class SearchQueryItemView extends RecyclerView.ViewHolder {
        TextView search_qeury_tv;

        public SearchQueryItemView(@NonNull final View itemView) {
            super(itemView);
            search_qeury_tv = (TextView) itemView.findViewById(R.id.search_query_tv);

            itemView.setOnClickListener(new View.OnClickListener(){
                @Override
                public void onClick(View v) {
                    int pos = getAdapterPosition();
                    if(pos != RecyclerView.NO_POSITION){
                        if(mListener != null){
                            mListener.onItemClick(v, pos);
                        }
                    }
                }
            });
        }
    }

}

File: app/src/main/java/com/example/todaysshow/adapter/SearchQueryAdapter.kt
package com.example.todaysshow.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Filter
import android.widget.Filterable
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.example.todaysshow.R
import com.example.todaysshow.TodayShowApplication

class SearchQueryAdapter(context: Context) : RecyclerView.Adapter<SearchQueryAdapter.SearchQueryHolder>(),
    Filterable {
    var context: Context = context
    var popularList: ArrayList<String> = TodayShowApplication.instance.popularSearchList
    var defaultList: ArrayList<String> = TodayShowApplication.instance.defaultSearchList
    var filteredList: ArrayList<String> = popularList
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchQueryHolder {
        val inflatedView = LayoutInflater.from(parent.context).inflate(R.layout.search_list_item, parent, false)
        return SearchQueryAdapter.SearchQueryHolder(inflatedView)
    }

    override fun onBindViewHolder(holder: SearchQueryHolder, position: Int) {
        val query : String = filteredList.get(position)
        holder.searchQueryTv.setText(query)
        holder.searchQueryTv.setOnClickListener(View.OnClickListener {
            Toast.makeText(context, query, Toast.LENGTH_SHORT).show()
        })
    }

    override fun getItemCount(): Int {
        return filteredList.size
    }

    override fun getFilter(): Filter {
       return object: Filter(){
           override fun performFiltering(constraint: CharSequence?): FilterResults {
               val charString: String = constraint.toString()
               if(charString.isEmpty()){
                   filteredList = popularList
               }else{
                   var filteringList : ArrayList<String> = ArrayList()
                   for (name in defaultList){
                       if(name.toLowerCase().contains(charString.toLowerCase())){
                           filteringList.add(name)
                       }
                   }
                   filteredList = filteringList
               }
               val filterResults : FilterResults = FilterResults()
               filterResults.values = filteredList
               return filterResults
           }

           override fun publishResults(constraint: CharSequence?, results: FilterResults?) {
               filteredList = results!!.values as ArrayList<String>
               notifyDataSetChanged()
           }
       }
    }

    class SearchQueryHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        var searchQueryTv = itemView?.findViewById<TextView>(R.id.search_query_tv)
        init {
            itemView.setOnClickListener(View.OnClickListener {
                var pos : Int = adapterPosition
                if(pos != RecyclerView.NO_POSITION){

                }
            })
        }
    }


}